File: week6/Week6Day4/app/src/main/java/com/example/luisenriquez/week6day4/data/repositorymodule/RepositoryImpl.java
package com.example.luisenriquez.week6day4.data.repositorymodule;

import android.annotation.SuppressLint;
import android.arch.lifecycle.MutableLiveData;

import com.example.luisenriquez.week6day4.data.remote.RemoteServiceHelper;
import com.example.luisenriquez.week6day4.data.remote.models.MovieResponse;

import java.util.List;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class RepositoryImpl implements Repository
{
    private MutableLiveData<MovieResponse> movieResponseLiveData = new MutableLiveData<>();
    private RemoteServiceHelper MyRemoteServiceHelper;

    public RepositoryImpl(RemoteServiceHelper InRemoteServiceHelper)
    {
        this.MyRemoteServiceHelper = InRemoteServiceHelper;
    }

    @SuppressLint("CheckResult")
    @Override
    public void getMovieResponse(String srchItm)
    {
        MyRemoteServiceHelper.getMovieResponse(srchItm)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(movieApiResponse ->
                                movieResponseLiveData.setValue(movieApiResponse),
                        Throwable::printStackTrace);
    }

    public MutableLiveData<MovieResponse> getMovieResponseLiveData()
    {
        return movieResponseLiveData;
    }

}

File: Week7/Week7Day4Test/app/src/main/java/com/example/luisenriquez/week7day4test/data/repositorymodule/RepositoryImpl.kt
package com.example.luisenriquez.week7day4test.data.repositorymodule

import android.arch.lifecycle.MutableLiveData
import android.util.Log
import com.example.luisenriquez.week7day4test.data.remote.RemoteServiceHelper
import com.example.luisenriquez.week7day4test.data.remote.model.NumbersResponse
import kotlinx.coroutines.experimental.CoroutineExceptionHandler
import kotlinx.coroutines.experimental.launch
import retrofit2.HttpException

class RepositoryImpl (private val remoteServiceHelper: RemoteServiceHelper) : Repository
{
    override val numberResponseLiveData: MutableLiveData<NumbersResponse> = MutableLiveData()

    override fun getRandomNumbers()
    {
        val exceptionHandler = CoroutineExceptionHandler {_,
                                                         throwable ->
            if (throwable is HttpException)
            {
                Log.d("main", "In HttpException")
            }
            else
            {
                Log.d("Main", "Other error: ", throwable.fillInStackTrace())
            }
        }

        launch (exceptionHandler)
        {
            val metaNumbersResponse = remoteServiceHelper.getRandomNumbers()
                                                            .await()
            numberResponseLiveData.postValue(metaNumbersResponse)
        }
    }
}